packages/react-native/Libraries/Blob/URLSearchParams.js

```diff
@@ -11,60 +11,138 @@
 // Small subset from whatwg-url: https://github.com/jsdom/whatwg-url/tree/master/src
 // The reference code bloat comes from Unicode issues with URLs, so those won't work here.
 export class URLSearchParams {
-  _searchParams: Array<[string, string]> = [];
+  _searchParams: Map<string, string[]> = new Map();
 
-  constructor(params?: Record<string, string>) {
-    if (typeof params === 'object') {
-      Object.keys(params).forEach(key => this.append(key, params[key]));
+  constructor(params?: Record<string, string> | string | [string, string][]) {
+    if (params === null) {
+      return;
+    }
+
+    // URLSearchParams("key1=value1&key2=value2");
+    if (typeof params === 'string') {
+      params
+        .replace(/^\?/, '')
+        .split('&')
+        .forEach(pair => {
+          if (!pair) {
+            return;
+          }
+          const [key, value] = pair
+            .split('=')
+            .map(part => decodeURIComponent(part.replace(/\+/g, ' ')));
+          this.append(key, value);
+        });
+    }
+
+    //URLSearchParams([["key1", "value1"], ["key2", "value2"]]);
+    else if (Array.isArray(params)) {
```

Comment:
Formatting nit (and below)
```suggestion
    } else if (Array.isArray(params)) {
      // URLSearchParams([["key1", "value1"], ["key2", "value2"]]);
```